### ISSUE TITLE
Deprecated guest with hacked url gets 502

### ISSUE THREAD

**Issue by robnagler** [radiasoft/sirepo#1923]
If a user tries to visit /auth-guest-login/elegant when guest is deprecated, the code throws an exception:

```txt
File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/sirepo/uri_router.py", line 82, in call_api
resp = flask.make_response(func(**kwargs) if kwargs else func())
File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/sirepo/auth/guest.py", line 43, in api_authGuestLogin
auth.complete_registration()
File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/sirepo/auth/__init__.py", line 140, in complete_registration
r = user_registration(u)
File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/sirepo/auth/__init__.py", line 401, in user_registration
res.save()
File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/sirepo/auth_db.py", line 54, in save
_db.session.commit()
File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 162, in do
return getattr(self.registry(), name)(*args, **kwargs)
```

**Comment by robnagler** [comment]
I was logged in with email at the time so that's a factor, too.

**Comment by robnagler** [comment]
Fix 7e80fe54
